crates/core/src/dec.rs

```diff
@@ -285,17 +335,18 @@ impl TryFrom<String> for Dec {
     }
 }
 
-impl From<Amount> for Dec {
-    fn from(amt: Amount) -> Self {
-        match I256::try_from(amt.raw_amount()).ok() {
-            Some(raw) => Self(
-                raw * Uint::exp10(
-                    (POS_DECIMAL_PRECISION - NATIVE_MAX_DECIMAL_PLACES)
-                        as usize,
-                ),
-            ),
-            None => Self::zero(),
-        }
+impl TryFrom<token::Amount> for Dec {
+    type Error = Error;
+
+    fn try_from(amt: token::Amount) -> std::result::Result<Self, Self::Error> {
+        let raw = I256::try_from(amt.raw_amount())
+            .map_err(|e| eyre!("Invalid raw amount: {e}"))?;
+        let denom = I256(Uint::exp10(
+            (POS_DECIMAL_PRECISION - NATIVE_MAX_DECIMAL_PLACES) as usize,
```

Comment:
I'm sure we're safe to do this operation but figured I'd point it out as possibly unchecked

Reply:
it's ok as they're both consts and the fn is also covered by unit tests